src/LinAlg/hiopVectorIntCuda.hpp

```diff
@@ -0,0 +1,118 @@
+//
+// Additional BSD Notice
+// 1. This notice is required to be provided under our contract with the U.S. Department 
+// of Energy (DOE). This work was produced at Lawrence Livermore National Laboratory under 
+// Contract No. DE-AC52-07NA27344 with the DOE.
+// 2. Neither the United States Government nor Lawrence Livermore National Security, LLC 
+// nor any of their employees, makes any warranty, express or implied, or assumes any 
+// liability or responsibility for the accuracy, completeness, or usefulness of any 
+// information, apparatus, product, or process disclosed, or represents that its use would
+// not infringe privately-owned rights.
+// 3. Also, reference herein to any specific commercial products, process, or services by 
+// trade name, trademark, manufacturer or otherwise does not necessarily constitute or 
+// imply its endorsement, recommendation, or favoring by the United States Government or 
+// Lawrence Livermore National Security, LLC. The views and opinions of authors expressed 
+// herein do not necessarily state or reflect those of the United States Government or 
+// Lawrence Livermore National Security, LLC, and shall not be used for advertising or 
+// product endorsement purposes.
+
+#pragma once
+
+/**
+ * @file hiopVectorIntCuda.hpp
+ *
+ * @author Nai-Yuan Chiang <chiang7@llnl.gov>, LLNL
+ */
+
+#include "hiopVectorInt.hpp"
+#include <string>
+
+namespace hiop
+{
+
+class hiopVectorIntCuda : public hiopVectorInt
+{
+private:
+  index_type *buf_host_;
+  index_type *buf_;
+  std::string mem_space_;
+
+public:
+  hiopVectorIntCuda(size_type sz, std::string mem_space="HOST");
+
+  ~hiopVectorIntCuda();
+
+  /**
+   * @brief Copy array data from the device.
+   *
+   * @note This is a no-op if the memory space is _host_ or _uvm_.
+   */
+  void copy_from_dev();
+
+  /**
+   * @brief Copy array data to the device.
+   *
+   * @note This is a no-op if the memory space is _host_ or _uvm_.
+   */
+  void copy_to_dev();
+
+  virtual inline index_type* local_data_host() { return buf_host_; }
+
+  virtual inline const index_type* local_data_host_const() const { return buf_host_; }
+
+  virtual inline index_type* local_data() { return buf_; }
+
+  virtual inline const index_type* local_data_const() const { return buf_; }
+  
+  virtual void copy_from(const index_type* v_local);
+
+  /// @brief Set all elements to zero.
+  virtual void set_to_zero();
+
+  /// @brief Set all elements  to  c
+  virtual void set_to_constant(const index_type c);
+
+  /**
+   * @brief Set the vector entries to be a linear space of starting at i0 containing evenly 
+   * incremented integers up to i0+(n-1)di, when n is the length of this vector
+   *
+   * @pre The elements of the linear space should not overflow the index_type type
+   *  
+   * @param i0 the starting element in the linear space (entry 0 in vector)
+   * @param di the increment for subsequent entries in the vector
+   *
+   */ 
```

Comment:
I mainly copy all the comments from other files, since no new functions have been introduced in this PR.
We may need to open a new PR to correct the comments in `hiopVectorInt.hpp`, `hiopVectorIntSeq.hpp`, `hiopVectorIntRaja.hpp` and this file. It seems identical comments should be added into these four files. 

Should we just add detailed comments in the abstract class, and use brief comments in the derived classes? @cnpetra

Reply:
Created issue #576